### PR TITLE
Adding logic for removing invalid views in compiler, and updated compiler tests. 

### PR DESCRIPTION
**Pull request by jaywoo123**
Adding code in Complier.py that adds a step for removing invalid views during view collection generation(ex. year by year). Adding old test cases back to test_compile.py and adding showMore calls to the test cases.

### CODE REVIEW

File: lux/compiler/Compiler.py [hunk 1]
```diff
@@ -111,6 +114,8 @@ def expandUnderspecified(ldf, viewCollection):
 		ldf : lux.luxDataFrame.LuxDataFrame
 			LuxDataFrame with underspecified context
 
+		viewCollection : list[lux.view.View]
+			empty list that will be populated with specified lux.View objects.
```
Comment (dorisjlee) [inline lux/compiler/Compiler.py:118]: This is not always empty right?

File: lux/compiler/Compiler.py [hunk 2]
```diff
@@ -136,6 +141,35 @@ def expandUnderspecified(ldf, viewCollection):
 						chartTitle = spec.value
 					view.title = f"{spec.attribute} {spec.filterOp} {chartTitle}"
 		return views
+
+	@staticmethod
+	def removeAllInvalid(viewCollection):
+		"""
+		Given an expanded view collection, remove all views that are invalid.
```
Comment (dorisjlee) [inline lux/compiler/Compiler.py:148]: Add an explanation regarding what types of views are considered invalid and what we are removing here.

Comment (jaywoo123) [reply lux/compiler/Compiler.py:148]: Added to a new commit.

File: lux/compiler/Compiler.py (hunk 1 above)
Comment (jaywoo123) [reply lux/compiler/Compiler.py]: Yeah, I think I blindly copy and pasted. Updated.

**Comment by dorisjlee** [comment]
This looks great, thanks @jaywoo123! I added some minor changes to reuse test_showMore to clean up the code. It should be ready to be merged!

**Pull request merged**